---
title: "first-draft"
author: "sourav-roy"
date: "2025-09-24"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

### Libraries

```{r}

# List all the libraries thal will be required for building the 
required_libraries <- c(
  "tidyverse",      # Data wrangling and visualization
  "DESeq2",         # Differential expression
  "edgeR",          # Alternative DE method
  "limma",          # Voom pipeline
  "tximport",       # Import transcript-level quantification
  "biomaRt",        # Gene annotation
  "GEOquery",       # Download GEO datasets
  "fgsea",          # Fast GSEA
  "msigdbr",        # MSigDB gene sets
  "GSEABase",       # Gene set manipulation
  "jsonlite",       # JSON support
  "ggplot2",        # Plotting
  "pheatmap",       # Simple heatmaps
  "cowplot",        # Plot composition
  "RColorBrewer",   # Color palettes
  "ComplexHeatmap"  # Advanced annotated heatmaps
)

# Load necessary libraries
library(GEOquery)
library(biomaRt)
library(tximport)
library(ggplot2)
```


```{r}
bulk_one <- "GSE249385"
bulk_two <- "GSE151387"
sc_one <- "GSE249558"
sc_two <- "GSE142089"

gset <- getGEO(sc_two, GSEMatrix = TRUE, getGPL = FALSE)
```


```{r}
sc_two_eset <- gset[[1]]
sc_two_metadata <- pData(sc_two_eset) # Contains the metadata we are looking for to identify the EpCam cells
```

